feat(site): Added json-ld for product data

tobi-or-not-tobi opened the pull request:
Added json-ld for product and product offers. 

Product data contains the basic product data and prices. The `AggregateOffer` is provided when the merchant package is installed. 

closes [HRZ-91023](https://spryker.atlassian.net/browse/HRZ-91023)

[HRZ-91023]: https://spryker.atlassian.net/browse/HRZ-91023?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

tobi-or-not-tobi committed:
spike

tobi-or-not-tobi committed:
paths

tobi-or-not-tobi committed:
fix import

tobi-or-not-tobi committed:
move jsonld to site and move logic to service

tobi-or-not-tobi committed:
paths

tobi-or-not-tobi committed:
eb

dunqan commented on libs/domain/site/src/services/jsonld/jsonld.normalizer.ts:
```diff
@@ -0,0 +1,21 @@
+import { Observable } from 'rxjs';
+import { JSONLD } from './jsonld.model';
+
+export interface JsonLdNormalizer<T = any> {
```
What about reusing existing transformer infra for it (what we use for adapter normalizers)?

dunqan committed:
Merge remote-tracking branch 'origin/development' into feature/product-json-ld

# Conflicts:
#	libs/domain/product/src/services/product.providers.ts

dunqan committed:
chore: json-ld for product data using transformers (#1074)

- refactor of initial jsonld transformer implementation
- support for class based transformers

dunqan committed:
Merge remote-tracking branch 'origin/development' into feature/product-json-ld

# Conflicts:
#	libs/domain/merchant/src/services/providers.ts

dunqan committed:
Merge remote-tracking branch 'origin/development' into feature/product-json-ld
